src/Sheet.php

```diff
@@ -248,8 +251,29 @@ public function import($import, int $startRow = 1)
 
         if ($import instanceof OnEachRow) {
             $headingRow = HeadingRowExtractor::extract($this->worksheet, $import);
+            $rows = [];
+
             foreach ($this->worksheet->getRowIterator()->resetStart($startRow ?? 1) as $row) {
-                $import->onRow(new Row($row, $headingRow));
+                $row = new Row($row, $headingRow);
+                $rows[$row->getIndex()] = $row;
```

Comment:
Hm, this kinda defeats the purpose of using the onEachRow. It will now load the entire worksheet into memory instead of using the iterator to save memory. I think I'd prefer if we'd call the validation method for each row individually instead of in batches.

Reply:
Are you thinking something more like:

```php
if ($import instanceof OnEachRow) {
    $headingRow = HeadingRowExtractor::extract($this->worksheet, $import);
    $rows = [];

    foreach ($this->worksheet->getRowIterator()->resetStart($startRow ?? 1) as $row) {
        $row = new Row($row, $headingRow);

		if ($import instanceof WithValidation) {
	        try {
	            app(RowValidator::class)->validate([$row->toArray(null, $import instanceof WithCalculatedFormulas)], $import);
	        } catch (RowSkippedException $e) {
	            continue;
	        }
	    }

	    $import->onRow($row);

        if ($import instanceof WithProgressBar) {
            $import->getConsoleOutput()->progressAdvance();
        }
    }
}
```